datafusion/src/optimizer/simplify_expressions.rs

```diff
@@ -1291,6 +1337,11 @@ mod tests {
 
     #[test]
     fn simplify_expr_case_when_then_else() {
+        // CASE WHERE c2 != false THEN "ok" == "not_ok" ELSE c2 == true
```

Comment:
Unfortunately the optimizer is not especially brilliant at optimizing boolean expressions - I've created #1716 to track this.

I'm not sure if we want to constrain the rewrite until the boolean expression optimizer is able to more effectively reduce down what it produces...

Reply:
🤔  I do wonder given the nullability concerns, if in addition to "simplify" which seeks to retain the same semantics, there could be something like:

```rust
/// returns true if this expr *always* returns null or false
/// (aka would filter out all rows from a query)
fn is_null_or_false(expr: &Expr) -> bool {
  ...
}
```

Reply:
I'm not sure I follow, are you suggesting I change something or just musing :sweat_smile:

Reply:
Sorry -- I was musing -- no specific changes suggested

Reply:
I implemented something related here: https://github.com/influxdata/influxdb_iox/pull/3557/files#r796984073